src/server.rs

```diff
@@ -10,8 +10,9 @@ use crate::store::Store;
 use crate::Error;
 
 pub async fn run(port: u16) -> Result<(), Error> {
-    let subscriber = tracing_subscriber::FmtSubscriber::new();
-    tracing::subscriber::set_global_default(subscriber)?;
+    let _ = tracing_subscriber::fmt()
+        .try_init()
+        .map_err(|e| debug!("Failed to initialize global tracing: {}", e));
```

Comment:
We had an issue where tests couldn’t run in parallel (Multiple calls to this run fn) because the global tracing subscriber was already set. This fix checks if the global config is already set and only sets it if it hasn’t been initialized.